nova/virt/vmwareapi/driver.py

```diff
@@ -1270,52 +1270,57 @@ def in_cluster_vmotion(self, context, instance, host_moref_value):
         We do not need to check if instance is part of our cluster, because we
         only get called if the instance belongs to us - in the DB at least.
         """
-        host_ref = vim_util.get_moref(host_moref_value, 'HostSystem')
-        try:
-            host_name = self._session._call_method(
-                vim_util, "get_object_property", host_ref, "name")
-        except vexc.ManagedObjectNotFoundException:
-            msg = f"Cannot find given target node {host_moref_value}"
-            raise error_util.InClustervMotionCheckError(reason=msg)
-        host_stats = self._vc_state.get_host_stats().get(host_name)
-        if not host_stats:
-            msg = (f"Cannot find given node {host_name} in host "
-                   "stats (doesn't belong to this cluster or is a "
-                   "failover host)")
-            raise error_util.InClustervMotionCheckError(reason=msg)
-        if not host_stats['available']:
-            msg = (f"Given node {host_name} is unsuitable "
-                   "(available = False)")
-            raise error_util.InClustervMotionCheckError(reason=msg)
-
-        vm_ref = vm_util.get_vm_ref(self._session, instance)
-        current_host_ref = self._session._call_method(vim_util,
-            "get_object_property", vm_ref, "runtime.host")
-        if vim_util.get_moref_value(current_host_ref) == host_moref_value:
-            LOG.debug("Target host %(host_moref_value)s is the current host",
-                      {'host_moref_value': host_moref_value},
-                      instance=instance)
-            return
+        # we cannot use wrap_instance_event() here, because that would change
+        # the public interface of our class which breaks testing ...
+        with compute_utils.EventReporter(context,
```

Comment:
Wouldn't  `@compute_utils.wrap_instance_event(prefix='vmware')`  do the trick? (After setting `self.host`   in `VMwareVCDriver.init_host`)

Reply:
I.e.:
```diff
class VMwareVCDriver
  ...

   def init_host(self, host):
+   self.host = host
...

+ @compute_utils.wrap_instance_event(prefix='vmware')
   def in_cluster_vmotion(self, context, instance, host_moref_value):
```

Reply:
No, as I stated in the comment, that breaks the driver's public interface, because the function then takes only `context` explicitly, which breaks the test. See https://github.com/sapcc/nova/blob/e675ba5a98d448519f244cec4e8516f84269dad3/nova/compute/utils.py#L1505 and https://ci1.eu-de-2.cloud.sap/builds/2514215075#L658b764a:17801

Reply:
I would say, the test is broken with respect to handling wrappers. One could also fix the test.
Just wanting to point out the options.

Reply:
Yes, what would you prefer?

Reply:
I think, keeping yours. While changing the test would be more "correct", it would make the SAP specific code change affect more places.